sympy/stats/frv_types.py

```diff
@@ -137,26 +156,30 @@ def check(sides):
                     "number of sides must be a positive integer.")
 
     @property
-    @cacheit
-    def dict(self):
-        as_int(self.sides) # Check that self.sides can be converted to an integer
-        return dict((k, Rational(1, self.sides)) for k in self.set)
+    def is_symbolic(self):
```

Comment:
If `is_number` can replace this, then let's use `is_number`. I would remove this method.

Reply:
Sure, I think mistook earlier review. I will remove this method.

Reply:
Well, removing this will not be possible because, it helps in generalising the check that whether a distribution has symbolic dimension or not. It has a default value of `False` in the super class and various distributions override according to the parameters they have. If I will remove this then it will invite a lot of code repetition. Though I will introduce the use of `is_number` in its definition.

Reply:
For example, in `HypergeometricDistribution` has a different for logic for checking whether it has symbolic dimension and `BinomialDistribution` has a different one. Now, it will not be possible to check that thing in the super class without introducing ad-hoc techniques which I do not support.
`is_symbolic` acts like a wrapper to that special logic and we can check that in the super class without much trouble.

However, I have removed `_is_sym_dim` because, `is_number` performs the same functionality.